pyanaconda/network.py

```diff
@@ -1390,120 +802,6 @@ def update_hostname_data(ksdata, hostname):
-                    log.debug("setting ONBOOT value of %s to %s", devname, network_data.onboot)
-                except (nm.SettingsNotFoundError, nm.UnknownDeviceError) as e:
-                    log.debug("setOnboot: %s", e)
-                    continue
-                except nm.MultipleSettingsFoundError:
-                    # In case of multiple connections for a device, update ifcfg directly
-                    if not update_onboot_value(devname, network_data.onboot, root_path=""):
-                        continue
-
-            updated_devices.append(devname)
-
-        if network_data.bondslaves or network_data.teamslaves or network_data.bridgeslaves:
-            updated_slaves = update_slaves_onboot_value(master, network_data.onboot)
-            updated_devices.extend(updated_slaves)
-
-    return updated_devices
-
-def apply_kickstart(ksdata):
-    applied_devices = []
-
-    for i, network_data in enumerate(ksdata.network.network):
-
-        # TODO: wireless not supported yet
-        if network_data.essid:
-            continue
-
-        if network_data.activate is None and i == 0:
-            network_data.activate = True
-
-        dev_name = get_device_name(network_data)
-        if not dev_name:
-            log.warning("apply kickstart: --device %s does not exist", network_data.device)
-            continue
-
-        ifcfg_path = find_ifcfg_file_of_device(dev_name)
-        if ifcfg_path:
-            if ifcfg_is_from_kickstart(ifcfg_path):
-                if network_data.activate:
-                    ensure_active_ifcfg_connection_for_device(ifcfg_path, dev_name)
-                continue
-
-        # If we don't have kickstart ifcfg from initramfs the command was added
-        # in %pre section after switch root, so apply it now
-        applied_devices.append(dev_name)
-        if ifcfg_path:
-            # if the device was already configured in initramfs update the settings
-            log.debug("pre kickstart - updating settings of device %s", dev_name)
-            con_uuid = update_settings_with_ksdata(dev_name, network_data)
-            added_connections = [(con_uuid, dev_name)]
-        else:
-            log.debug("pre kickstart - adding connection for %s", dev_name)
-            # Virtual devices (eg vlan, bond) return dev_name == None
-            added_connections = add_connection_for_ksdata(network_data, dev_name)
-
-        if network_data.activate:
-            for con_uuid, dev_name in added_connections:
-                try:
-                    log.debug("pre kickstart - activating connection %s for %s", con_uuid, dev_name)
-                    nm.nm_activate_device_connection(dev_name, con_uuid)
-                except (nm.UnknownConnectionError, nm.UnknownDeviceError) as e:
-                    log.warning("pre kickstart: can't activate connection %s on %s: %s",
-                                con_uuid, dev_name, e)
-    return applied_devices
-
 def networkInitialize(ksdata):
```

Comment:
I think that the network initialization should handled via a DBus task or a list of DBus tasks.

Reply:
I'll think about it. I think I'd prefer a list of tasks. One thing to consider is that now network initialization is not run in thread. I should review some NM asynchronous methods we are calling now without syncing on result (as we used to) while on that.
Something for a separate PR.

Reply:
You can use sync_run_task to wait for the tasks to finish for now.